// Code generated by ent, DO NOT EDIT.

package ent

import (
	"context"
	"go-manage/pkg/ent/personal"
	"go-manage/pkg/ent/predicate"

	"entgo.io/ent/dialect/sql"
	"entgo.io/ent/dialect/sql/sqlgraph"
	"entgo.io/ent/schema/field"
)

// PersonalDelete is the builder for deleting a Personal entity.
type PersonalDelete struct {
	config
	hooks    []Hook
	mutation *PersonalMutation
}

// Where appends a list predicates to the PersonalDelete builder.
func (pd *PersonalDelete) Where(ps ...predicate.Personal) *PersonalDelete {
	pd.mutation.Where(ps...)
	return pd
}

// Exec executes the deletion query and returns how many vertices were deleted.
func (pd *PersonalDelete) Exec(ctx context.Context) (int, error) {
	return withHooks[int, PersonalMutation](ctx, pd.sqlExec, pd.mutation, pd.hooks)
}

// ExecX is like Exec, but panics if an error occurs.
func (pd *PersonalDelete) ExecX(ctx context.Context) int {
	n, err := pd.Exec(ctx)
	if err != nil {
		panic(err)
	}
	return n
}

func (pd *PersonalDelete) sqlExec(ctx context.Context) (int, error) {
	_spec := sqlgraph.NewDeleteSpec(personal.Table, sqlgraph.NewFieldSpec(personal.FieldID, field.TypeInt))
	if ps := pd.mutation.predicates; len(ps) > 0 {
		_spec.Predicate = func(selector *sql.Selector) {
			for i := range ps {
				ps[i](selector)
			}
		}
	}
	affected, err := sqlgraph.DeleteNodes(ctx, pd.driver, _spec)
	if err != nil && sqlgraph.IsConstraintError(err) {
		err = &ConstraintError{msg: err.Error(), wrap: err}
	}
	pd.mutation.done = true
	return affected, err
}

// PersonalDeleteOne is the builder for deleting a single Personal entity.
type PersonalDeleteOne struct {
	pd *PersonalDelete
}

// Where appends a list predicates to the PersonalDelete builder.
func (pdo *PersonalDeleteOne) Where(ps ...predicate.Personal) *PersonalDeleteOne {
	pdo.pd.mutation.Where(ps...)
	return pdo
}

// Exec executes the deletion query.
func (pdo *PersonalDeleteOne) Exec(ctx context.Context) error {
	n, err := pdo.pd.Exec(ctx)
	switch {
	case err != nil:
		return err
	case n == 0:
		return &NotFoundError{personal.Label}
	default:
		return nil
	}
}

// ExecX is like Exec, but panics if an error occurs.
func (pdo *PersonalDeleteOne) ExecX(ctx context.Context) {
	if err := pdo.Exec(ctx); err != nil {
		panic(err)
	}
}
